Bobsar0/Fast-Food-Fast
# #161298509 Consume placeorder endpoint

What does this PR do?
Consumes POST /orders endpoint using Fetch API

Description of Task to be completed?
Consume place order endpoint so users can successfully place immediate or add-to-cart orders.
This is achieved via the following tasks:
- modify ordersController to process orders with multiple food items
- implement placing of multiple-item (add to cart) orders
- add buy now feature to menu for single item immediate purchase
- modify single-item checkout feature
- implement single-item order processing


How should this be manually tested?
Clone the repo and run the following commands:
```
  - git checkout ft-consume-placeorder-endpoint-#161298509
  - npm install
  - npm start
```
Navigate to http://localhost:9999/ and login/create an account.
You should be redirected to a dedicated menu page
For immediate single-item purchase - click `Buy now` on any food item, enter delivery address and phone number and submit order. 
For add-to-cart cart purchases, click on `Add to Cart` on one or more items, perform the following:
```
- Click on the cart icon on the navigation bar
- Enter delivery address and phone number
- Peruse the order and delete any item not needed
- Submit order
```
If successful on both cases, you should see an order confirmation page with details of your order.
Otherwise an error descriptive message will appear (for eg. `item x not available in stock`)

What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/#161298509
